### DISCUSSION TITLE
How to get the Dynamic Utility value as the class suffix

### DISCUSSION THREAD

**Discussion by thanveershah** [tailwindlabs/tailwindcss#8068] [Help] [answered]
Example: 

```html
<div class="pl-10" />
```

Currently, I can add static class names 

```js
addUtilities({
      ".pl-10": {
         "padding-inline-start": "10px",
      },
   });
```

I want to get the value given by the user without hard coding it, like below
```js
addUtilities({
     ".pl-[value]": {
        "padding-inline-start": value
      },
   });
```

**Comment by mubaraqwahab** [comment, marked as answer]
You can use [`matchUtilities`](https://tailwindcss.com/docs/plugins#dynamic-utilities). Something like this perhaps (haven't tested it):

```js
plugin(function({ matchUtilities, theme }) {
      matchUtilities(
        {
          pl: (value) => ({
            paddingInlineStart: value
          }),
        },
        { values: theme('padding') }
      )
    })
```

This way, you can use `.pl-` with the existing `padding` values in your theme as well as with arbitrary values.

**Reply by thanveershah** [reply]
Awsome, thanks man !
